src/poly/commitment.rs

```diff
@@ -154,6 +245,52 @@ impl<C: CurveAffine> Params<C> {
     }
 }
 
+/// A guard returned by the verifier
+#[derive(Debug)]
+pub struct Guard<C: CurveAffine> {
+    msm: MSM<C>,
+    neg_z1: C::Scalar,
+    allinv: C::Scalar,
+    challenges_sq: Vec<C::Scalar>,
+    challenges_sq_packed: Vec<Challenge>,
+}
+
+impl<C: CurveAffine> Guard<C> {
+    /// Lets caller supply the challenges and obtain an MSM with updated
+    /// scalars and points.
+    pub fn use_challenges(mut self, params: &Params<C>) -> Result<MSM<C>, Error> {
+        let mut scalars: Vec<C::Scalar> = vec![];
+        let mut bases: Vec<C> = vec![];
+
+        scalars.extend(&self.msm.other_scalars);
+        bases.extend(&self.msm.other_bases);
+
+        // - [z2] H
+        if let Some(h_scalar) = self.msm.h_scalar {
```

Comment:
This operation is unnecessary because the commitment verifier will emplace the correct `h_scalar` into the `MSM` for us.